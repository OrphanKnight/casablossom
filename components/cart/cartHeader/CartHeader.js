import styles from "./styles.module.scss";
import { useState, useEffect, useCallback } from "react";
import { useSelector, useDispatch } from "react-redux";
import { compareArrays } from "./../../../utils/arrays_utils";
import { toggleAllSelected } from "./../../../store/cartSlice";
export default function CartHeader() {
  const { cart } = useSelector((state) => ({ ...state }));
  const cartItems = cart.cartItems;
  const selected = cartItems.filter((item) => item.selected);
  const [active, setActive] = useState();
  const dispatch = useDispatch();
  const memoizedCompareArrays = useCallback(compareArrays, []);

  useEffect(() => {
    const check = memoizedCompareArrays(cartItems, selected);

    setActive(check);
  }, [cart.cartItems]);

  const handleSelect = () => {
    if (selected.length !== cartItems.length) {
      dispatch(toggleAllSelected(true));
    } else {
      dispatch(toggleAllSelected(false));
    }
  };
  return (
    <div className={`${styles.cart__header} ${styles.card}`}>
      <h1>Item Summary({cartItems.length})</h1>
      <div className={styles.flex} onClick={() => handleSelect()}>
        <div
          data-testid="checkbox"
          className={`${styles.checkbox} ${active ? styles.active : ""}`}
        ></div>
        <span>Select all items</span>
      </div>
    </div>
  );
}
