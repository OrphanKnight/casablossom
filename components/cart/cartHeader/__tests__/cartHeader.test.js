import { render } from '@testing-library/react';
import CartHeader from '../index';

it('renders CartHeader component', () => {
  const { container } = render(<CartHeader/>);
  expect(container).toMatchSnapshot();
})