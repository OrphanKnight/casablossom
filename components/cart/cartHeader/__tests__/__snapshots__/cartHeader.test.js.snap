// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CartHeader renders CartHeader component 1`] = `
<div>
  <div
    class="cart__header card"
  >
    <h1>
      Item Summary(
      0
      )
    </h1>
    <div
      class="flex"
    >
      <div
        class="checkbox active"
        data-testid="checkbox"
      />
      <span>
        Select all items
      </span>
    </div>
  </div>
</div>
`;
